nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -1002,61 +1004,62 @@ def write_units(
-                    if isinstance(prop_value, (int, np.integer, float, str, bool)):
-                        shapes.append(1)
-                    elif isinstance(prop_value, (list, np.ndarray)):
-                        if np.array(prop_value).ndim == 1:
-                            shapes.append(len(prop_value))
-                        else:
-                            shapes.append(np.array(prop_value).shape)
-                    elif isinstance(prop_value, dict):
-                        print(f"Skipping property '{pr}' because dictionaries are not supported.")
-                        skip_properties.append(pr)
-                        break
-                else:
-                    shapes.append(np.nan)
-            property_shapes[pr] = shapes
-
-        for pr in property_shapes.keys():
-            elems = [elem for elem in property_shapes[pr] if not np.any(np.isnan(elem))]
-            if not np.all([elem == elems[0] for elem in elems]):
-                print(f"Skipping property '{pr}' because it has variable size across units.")
-                skip_properties.append(pr)
-
-        write_properties = set(all_properties) - set(skip_properties)
-        for pr in write_properties:
-            if pr not in property_descriptions:
-                warnings.warn(
-                    f"Description for property {pr} not found in property_descriptions. "
-                    "Setting description to 'no description'"
-                )
-        for pr in write_properties:
-            unit_col_args = dict(name=pr, description=property_descriptions.get(pr, "No description."))
-            if pr in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
-                unit_col_args.update(table=nwbfile.electrodes)
-            nwbfile.add_unit_column(**unit_col_args)
+    units_table = pynwb.misc.Units(name=units_name, description=units_description)
 
+    # Check that array properties have the same shape across units
+    property_shapes = dict()
+    for pr in all_properties:
+        shapes = []
         for unit_id in unit_ids:
-            unit_kwargs = dict()
-            if use_times:
-                spkt = sorting.frame_to_time(sorting.get_unit_spike_train(unit_id=unit_id))
+            if pr in sorting.get_unit_property_names(unit_id):
+                prop_value = sorting.get_unit_property(unit_id, pr)
+                if isinstance(prop_value, (int, np.integer, float, str, bool)):
+                    shapes.append(1)
+                elif isinstance(prop_value, (list, np.ndarray)):
+                    if np.array(prop_value).ndim == 1:
+                        shapes.append(len(prop_value))
+                    else:
+                        shapes.append(np.array(prop_value).shape)
+                elif isinstance(prop_value, dict):
+                    print(f"Skipping property '{pr}' because dictionaries are not supported.")
+                    skip_properties.append(pr)
+                    break
```

Comment:
This appears to me to be a completely different type of change that could be separated from the write location change

Reply:
No changes other than the units table writing were included in this PR - this appears to be an artifact of the removal of [line #1005 from master](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/nwb_conversion_tools/utils/spike_interface.py#L1005), which re-nests the entire code block below it, which GitHub has trouble parsing in the changelog for whatever reason.

It's easier to see this if you put these code blocks from [master](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/nwb_conversion_tools/utils/spike_interface.py#L1005-L1059) and [units_provenance](https://github.com/catalystneuro/nwb-conversion-tools/blob/units_provenance/nwb_conversion_tools/utils/spike_interface.py#L1009-L1062) side by side for comparison.